### PR TITLE
add Linux command line interface for batch processing

### PR DESCRIPTION
**Pull request by danchal**
Please find a Linux equivalent for encrypt.bat and process.bat for your review.
Cheers,
Dan

### CODE REVIEW

File: src/main/nsis/bin/process.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
+#/bin/bash
+# change to directory of this script
+cd "$( dirname "${BASH_SOURCE[0]}" )"
```
Comment (cbhacking) [inline src/main/nsis/bin/process.sh]: The Windows scripts don't change the working directory. I can see why you're doing it - so the relative path to the .JAR is correct - but that could be better handled with a local variable.

Comment (danchal) [reply src/main/nsis/bin/process.sh]: Thanks for the review, you are correct, the behaviour must be the same as the Windows scripts. I will modify the PR.

**Comment by danchal** [comment]
I updated the [PR](https://github.com/forcedotcom/dataloader/pull/145/commits/47a327d63cc27d57381718f01c6c2a84f3cd9470), removing the change program directory lines. The behaviour is now the same as the Windows scripts.

Reading your last comment, are you suggesting that both the Windows and Linux scripts be now modified to handle the .JAR path better? I think that this would be an improvement on the current behaviour, if implemented as you have suggested.

**Comment by ashitsalesforce** [comment]
While we do not support linux CLI, you may be able to do so in v52 onwards (or try it in v51 if you compile from the source) by using the following options:

`java -jar dataloader-51.0.1-uber.jar run.mode=batch process.name=<process name> salesforce.config.dir=<path to config dir containing process-conf.xml and config.properties files>`

**Pull request closed without merging**